### PR TITLE
fix: main layout header shift

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Fix the layout shift by waiting for data to load before rendering.

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### CODE REVIEW

**Review by omBratteng — APPROVED** [review]

File: packages/shared/src/components/layout/MainLayoutHeader.tsx [hunk 1]
```diff
@@ -86,10 +86,63 @@ function MainLayoutHeader({
       extra: JSON.stringify({ notifications_number: unreadCount }),
     });
   };
-  const { referralToken } = useReferralCampaign({
+  const { referralToken, isReady } = useReferralCampaign({
```
Comment (rebelchris) [inline packages/shared/src/components/layout/MainLayoutHeader.tsx:89]: This also works when user is not eliged for it? (just making sure)

**Review by rebelchris — APPROVED** [review]
One check, but if that's the case we can go :)

File: packages/shared/src/components/layout/MainLayoutHeader.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/layout/MainLayoutHeader.tsx:89]: Yes, I tried incognito and it still renders the UI.

Comment (sshanzel) [reply packages/shared/src/components/layout/MainLayoutHeader.tsx:89]: Also validated from the docs.

Screenshot 2023-10-24 at 12 57 22 AM

**Pull request merged**